### PR TITLE
feat(sample): samples of animation

### PR DESCRIPTION
**Pull request by hellmor**
Add samples to show how to use skeletal animation and property animation

### CODE REVIEW

File: samples/animation/Sample_CurveAnimation.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    curve1: AnimationCurve;
+    curve2: AnimationCurve;
+    curve3: AnimationCurve;
+    curve4: AnimationCurve;
+
+    async run() {
+        await Engine3D.init({ beforeRender: () => this.renderUpdate() });
+
+        Engine3D.setting.shadow.shadowBound = 1000;
+        Engine3D.setting.shadow.shadowBias = 0.002;
```
Comment (lslzl3000) [inline samples/animation/Sample_CurveAnimation.ts:16]: better set 0.0001

File: samples/animation/Sample_Skeleton.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    lightObj3D: Object3D;
+    scene: Scene3D;
+    async run() {
+
+        Engine3D.setting.material.materialDebug = false;
+        Engine3D.setting.material.materialChannelDebug = true;
+        Engine3D.setting.shadow.autoUpdate = true;
+        Engine3D.setting.shadow.updateFrameRate = 2;
+        Engine3D.setting.shadow.shadowBound = 200;
+        Engine3D.setting.shadow.shadowBias = 0.00066;
```
Comment (lslzl3000) [inline samples/animation/Sample_Skeleton.ts:15]: set to 0.0001

File: samples/animation/Sample_Skeleton2.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+class Sample_Skeleton2 {
+    lightObj3D: Object3D;
+    scene: Scene3D;
+
+    async run() {
+
+        Engine3D.setting.shadow.autoUpdate = true;
+        Engine3D.setting.shadow.updateFrameRate = 2;
+        Engine3D.setting.shadow.shadowBound = 1000
+        Engine3D.setting.shadow.shadowBias = 0.002;
```
Comment (lslzl3000) [inline samples/animation/Sample_Skeleton2.ts:14]: same~ 0.0001

File: samples/animation/Sample_Skeleton3.ts [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+// Sample to use SkeletonAnimationComponent
+class Sample_Skeleton3 {
+    lightObj3D: Object3D;
+    scene: Scene3D;
+
+    async run() {
+        Engine3D.setting.shadow.autoUpdate = true;
+        Engine3D.setting.shadow.updateFrameRate = 1;
+        Engine3D.setting.shadow.shadowBound = 350
+        Engine3D.setting.shadow.shadowBias = 0.002;
```
Comment (lslzl3000) [inline samples/animation/Sample_Skeleton3.ts:14]: ~

**Review by Codeboy-cn — APPROVED** [review]

**Pull request merged**